cmd/update-rules/main.go

```diff
@@ -142,7 +146,9 @@ func UpdateRules(rules *config.RepositoryRules, branch, goVer string) {
 		}
 
 		// update the rules for branch and its dependencies
-		updateBranchRules(&newBranchRule, branch, goVer)
+		if !removeRules {
+			updateBranchRules(&newBranchRule, branch, goVer)
+		}
 
 		var branchRuleExists bool
 		// if the target branch rules already exists, update it
```

Comment:
Shouldnt this be also part of the `!removeRules` code block? Because this section needs to be executed only if we are not removing the rules, otherwise we will be updating it first and then immediately deleting it.